Library/Homebrew/cask/lib/hbc/cli/internal_stanza.rb

```diff
@@ -68,10 +65,18 @@ def run
             next
           end
 
-          if artifact_name && !value.key?(artifact_name)
-            opoo "no such stanza '#{artifact_name}' on Cask '#{cask}'" unless quiet?
-            puts ""
-            next
+          if value.nil? || (value.respond_to?(:to_a) && value.to_a.empty?) ||
+             (value.respond_to?(:to_h) && value.to_h.empty?) ||
```

Comment:
The `Enumerable#to_h` method, which was introduced in Ruby 2.1, **will fail unless every element it contains is a two-element array.** This appears to be why `#to_h` currently fails for two of the test cases.

For example, it fails when `value` is a `Set` whose elements are `Artifact`s (and not pairs of something); see also @reitermarkus’s [comment](https://github.com/Homebrew/brew/pull/3258/commits/a605d74e6521181f3a0c2e46de95abadab01320f#r143356789).

Fortunately, you should be able to just lose that line altogether. It is redundant because whenever a hash is empty, its `#to_a` is also empty.

Reply:
@claui I removed the line as you suggested. The test is still failing on TravisCI and is still working fine on my machine. Any ideas what I can do to get the tests failing on my machine as well in order to speed up fixing this?

Reply:
@Git-Jiro Run `brew tests --online` and `brew style`.

Reply:
@MikeMcQuaid unfortunately the --online flag did not change the results: https://gist.github.com/Git-Jiro/2afc9133c700223fd67383030142f9ed

Reply:
@Git-Jiro Can you double-check that you have actually tapped into `caskroom/cask` on the system you use for testing? Maybe `brew untap caskroom/cask && brew tap caskroom/cask` for good measure? (This shouldn’t normally be necessary.)

Reply:
Nope, tests are still fine on my system ...

Reply:
What happens if you drop a `raise` somewhere in the code under test?

Reply:
If I drop a raise in my code under test, the raise works as expected:
https://gist.github.com/Git-Jiro/9254614151ac0a4809ed20a5e70b422d

Reply:
Now that you find the `raise` to work as expected, try moving the `raise` statement around in the code. Move it gradually near the `if stanza == artifacts` block (this is the region of the code where the test fails), and repeat this step until the `raise` ends up inside that block. If the `raise` stops working at some point, then find out why. Alternatively, use debug statements like `warn` to figure out what’s happening near the offending part of the code.

As you can see in the CI failures, the `if stanza == artifacts` block is one of the reasons why the tests are still failing. (This is most likely a side effect of the recent refactoring of the `#artifacts` method, which now returns a `Set` instead of a `Hash`.)

Also, try to stick with [your `brew tests` command line](https://gist.github.com/Git-Jiro/2afc9133c700223fd67383030142f9ed) instead of running `brew cask _stanza`.

Reply:
Finally, I think I got all your hints about Sets vs. Hashes.
Hopefully the code works now ...